### PR TITLE
C library:  Add some new functions defined in POSIX.1-2017

### PR DESCRIPTION
**Pull request by thapav**
This patch adds following API's to libc:
stpncpy()
strsignal()
psignal()
psiginfo()

Back ported from NuttX

Signed-off-by: Vidisha Thapa <thapa.v@samsung.com>

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [5b217a63f57dc36df232eab01e0115ba29424384] - Code Rule Check Result:
lib/libc/string/lib_strsignal.c:70: ERROR: [BRC_M_SMT] that open brace { should be on the previous line

**Comment by seinfra** [comment]
Target : [5b217a63f57dc36df232eab01e0115ba29424384] - Code Rule Check (C++) OK.

File: os/include/string.h [hunk 1]
```diff
@@ -258,6 +265,13 @@ FAR char *strstr(FAR const char *, FAR const char *);
  * @since TizenRT v1.0
  */
 FAR char *strcasestr(FAR const char *, FAR const char *);
+/**
+ * @brief get name of signal
+ * @details @b #include <string.h> \n
+ * POSIX API (refer to : http://pubs.opengroup.org/onlinepubs/9699919799/)
+ * @since TizenRT v1.0
```
Comment (jeongchanKim) [inline os/include/string.h]: Please change the version to 2.0

File: os/include/string.h [hunk 2]
```diff
@@ -201,6 +201,13 @@ FAR char *strcpy(char *dest, FAR const char *src);
  * @since TizenRT v1.0
  */
 FAR char *stpcpy(FAR char *dest, FAR const char *src);
+/**
+ * @brief copy fixed length string, returning a pointer to the array end
+ * @details @b #include <string.h> \n
+ * POSIX API (refer to : http://pubs.opengroup.org/onlinepubs/9699919799/)
+ * @since TizenRT v1.0
```
Comment (jeongchanKim) [inline os/include/string.h]: Please change the version to 2.0

File: os/include/signal.h [hunk 3]
```diff
@@ -358,6 +358,14 @@ extern "C" {
  */
 int kill(pid_t pid, int sig);
 
+/**
+ * @brief print signal message
+ * @details @b #include <signal.h> \n
+ * POSIX API (refer to : http://pubs.opengroup.org/onlinepubs/9699919799/)
+ * @since TizenRT v2.0
```
Comment (jeongchanKim) [inline os/include/signal.h]: Please change the version to 2.1 PRE

File: os/include/string.h [hunk 4]
```diff
@@ -201,6 +201,13 @@ FAR char *strcpy(char *dest, FAR const char *src);
...
+ */
+FAR char  *stpncpy(FAR char *, FAR const char *, size_t);
```
Comment (sunghan-chang) [inline os/include/string.h]: Could you fill name of arguments? That is an our coding rule.

**Comment by sunghan-chang** [comment]
@thapav And please find a coding rule violation as shown above.

File: os/include/string.h (hunk 4 above)
Comment (thapav) [reply os/include/string.h]: @sunghan-chang 
There are a variety of argument names used for cpy API's
Is the below declaration fine ??
FAR char  *stpncpy(FAR char *s1, FAR const char *s2, size_t n);

File: lib/libc/string/lib_stpncpy.c [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+ *   undefined.
+ *
+ * Returned Value:
+ *   If a NUL character is written to the destination, the stpncpy()
+ *   function will return the address of the first such NUL character.
+ *   Otherwise, it will return &dest[n]
+ *
+ ****************************************************************************/
+
+#ifndef CONFIG_LIBC_ARCH_STPNCPY
```
[the remark this answers is not in the document]
Comment (thapav) [reply lib/libc/string/lib_stpncpy.c]: done

File: os/include/signal.h (hunk 3 above)
Comment (thapav) [reply os/include/signal.h]: done

File: os/include/string.h (hunk 2 above)
Comment (thapav) [reply os/include/string.h]: done

File: os/include/string.h (hunk 1 above)
Comment (thapav) [reply os/include/string.h]: done

**Comment by seinfra** [comment]
Target : [fe7bb193bf742f3364e28c2bebfe8ff9504989ac] - Code Rule Check Result:
lib/libc/string/lib_strsignal.c:70: ERROR: [BRC_M_SMT] that open brace { should be on the previous line

**Comment by seinfra** [comment]
Target : [fe7bb193bf742f3364e28c2bebfe8ff9504989ac] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [98f72e786dab4672f3ccd0dc720e966284122ec3] - Code Rule Check (C++) OK.

File: lib/libc/string/lib_stpncpy.c (hunk 5 above)
Comment (yashwanth686007) [inline lib/libc/string/lib_stpncpy.c]: Change CONFIG_LIBC_ARCH_STPNCPY -> CONFIG_ARCH_STPNCPY

Comment (thapav) [reply lib/libc/string/lib_stpncpy.c]: done

File: os/include/signal.h [hunk 6]
```diff
@@ -358,6 +358,14 @@ extern "C" {
  */
 int kill(pid_t pid, int sig);
 
+/**
```
[the remark this answers is not in the document]
Comment (thapav) [reply os/include/signal.h]: done

File: lib/libc/signal/sig_psignal.c [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+ *     indicated by pinfo will be written, followed by a newline.
+ *
+ * Returned Value
+ *  None.  Since no value is returned, an application wishing to check for
+ *  error situations should set errno to 0, then call psiginfo() then check
+ *  errno.
+ *
+ ****************************************************************************/
+
+void psiginfo(FAR const siginfo_t *pinfo, FAR const char *message)
```
[the remark this answers is not in the document]
Comment (thapav) [reply lib/libc/signal/sig_psignal.c]: done

**Comment by seinfra** [comment]
Target : [262e1faca359cc54c7ee6e74512967f22b4277e9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [262e1faca359cc54c7ee6e74512967f22b4277e9] - Code Rule Check (C++) OK.

File: os/include/string.h [hunk 8]
```diff
@@ -258,6 +265,13 @@ FAR char *strstr(FAR const char *, FAR const char *);
  * @since TizenRT v1.0
  */
 FAR char *strcasestr(FAR const char *, FAR const char *);
+/**
+ * @brief get name of signal
+ * @details @b #include <string.h> \n
+ * POSIX API (refer to : http://pubs.opengroup.org/onlinepubs/9699919799/)
+ * @since TizenRT v2.1 PRE
+ */
+FAR char  *strsignal(int signum);
```
Comment (sunghan-chang) [inline os/include/string.h]: Not important but only this has two spaces between return type and function pointer.

**Comment by sunghan-chang** [comment]
I left my comment again. Could you check it?
```
@thapav Could you split this as two commits? Please split changes on defconfig as a different commit.
```

File: os/include/string.h (hunk 8 above)
Comment (thapav) [reply os/include/string.h]: removed extra space

**Comment by seinfra** [comment]
Target : [269b1617f6e083a8446b5ed47a6aab49d686e44b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [269b1617f6e083a8446b5ed47a6aab49d686e44b] - Code Rule Check (C++) OK.

**Comment by thapav** [comment]
@sunghan-chang split it into two different commits..
Please review..

File: os/include/string.h [hunk 9]
```diff
@@ -201,6 +201,13 @@ FAR char *strcpy(char *dest, FAR const char *src);
  * @since TizenRT v1.0
  */
 FAR char *stpcpy(FAR char *dest, FAR const char *src);
+/**
```
Comment (jeongchanKim) [inline os/include/string.h]: You should add @cond and @internal, because there is no TC for this API.

File: os/include/string.h [hunk 10]
```diff
@@ -258,6 +265,13 @@ FAR char *strstr(FAR const char *, FAR const char *);
  * @since TizenRT v1.0
  */
 FAR char *strcasestr(FAR const char *, FAR const char *);
+/**
```
Comment (jeongchanKim) [inline os/include/string.h]: Same as above

**Comment by seinfra** [comment]
Target : [73d043b72dcf6f0e82293d8b724fda3b5fd34a28] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [73d043b72dcf6f0e82293d8b724fda3b5fd34a28] - Code Rule Check (C++) OK.

**Pull request merged**